hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -199,25 +201,63 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       setHmsTableParameters(newMetadataLocation, tbl, metadata.properties(), removedProps, hiveEngineEnabled);
 
       persistTable(tbl, updateHiveTable);
-      threw = false;
     } catch (org.apache.hadoop.hive.metastore.api.AlreadyExistsException e) {
-      throw new AlreadyExistsException("Table already exists: %s.%s", database, tableName);
+      commitFailed = true;
+      throw new CommitFailedException("Table already exists: %s.%s", database, tableName);
 
     } catch (TException | UnknownHostException e) {
       if (e.getMessage() != null && e.getMessage().contains("Table/View 'HIVE_LOCKS' does not exist")) {
+        commitFailed = true;
         throw new RuntimeException("Failed to acquire locks from metastore because 'HIVE_LOCKS' doesn't " +
```

Comment:
This should be a CFE too

Reply:
I am not entirely sure. Thrift client might throw exception even if the HMS server side is finished but the response did not arrive to the HMS client.

Reply:
I meant for this specific message, for the general case we don't know but if we get the Hive Locks error doesn't that mean we know we couldn't commit?